tox.ini

```diff
@@ -37,6 +37,8 @@ deps =
     lint,tests: pytest-mock
     lint,tests,functests: pytest
     lint,tests,functests: h-testkit
+    lint,functests: pytest-httpx
+    lint,functests: freezegun
```

Comment:
If you run `make template` you'll see that it removes these two lines from `tox.ini`. To fix this you need to create a `.cookiecutter/includes/tox/deps` file with these contents:

```
lint,functests: pytest-httpx
lint,functests: freezegun
```

Now `make template` won't delete these lines from `tox.ini` anymore.

The reason for this is that `make template` overwrites `tox.ini` with the result of rendering [a `tox.ini` template file](https://github.com/hypothesis/cookiecutters/blob/main/_shared/project/tox.ini). You'll see that the `deps` section in that `tox.ini` template includes [this](https://github.com/hypothesis/cookiecutters/blob/819732a9422e241bfe58afea1d3b6cc29a3ab727/_shared/project/tox.ini#L86-L88):

```jinja2
{% if include_exists("tox/deps") %}
    {{- include("tox/deps", indent=4) -}}
{% endif %}
```

That means that if the project has a `.cookiecutter/includes/tox/deps` file its contents will be included at that location in the `tox.ini`. It's a bit awkward, but this is how we enable projects to customize parts of templated files like `tox.ini` while still allowing the cookiecutter to control most of the contents of that file. And you only have to do this for cookiecutter-managed files (things that are largely the same in each project like `tox.ini`, `Makefile`, etc). The `include()` function is implemented [here](https://github.com/hypothesis/cookiecutters/blob/819732a9422e241bfe58afea1d3b6cc29a3ab727/_shared/local_extensions.py#L156-L160). There's a [presentation about our cookiecutters solution](https://docs.google.com/presentation/d/1bfAkyJvIQINusmYnw0tCH1ldlJ-ZDWd78o9VKjmulys/edit#slide=id.p) if you're interested in reading more about it.

Reply:
That's good to know and thank you for the presentation!